zilliqa/src/consensus.rs

```diff
@@ -379,9 +380,51 @@ impl Consensus {
             )?;
         }
 
+        // If timestamp of when current view was written exists then use it to estimate the minimum number of blocks the network has moved on since shut down
+        // This is useful in scenarios in which consensus has failed since this node went down
+        if let Some(latest_view_timestamp) = consensus.db.get_view_updated_at()? {
+            let view_diff = Consensus::minimum_views_in_time_difference(
+                latest_view_timestamp.elapsed()?,
+                consensus.config.consensus.consensus_timeout,
+            );
+            let start_view_jumped_ahead = start_view + view_diff;
+            consensus.db.set_view(start_view_jumped_ahead)?;
+            debug!(
+                "Atleast {} views changed since last view was written. new start_view {}",
+                view_diff, start_view_jumped_ahead
+            );
+
+            // Build a NewView message ready for sending for the view we jump into. If other nodes are still on this view then this will allow for consensus to resume immediately.
```

Comment:
Fair. I've removed the NewView store.

We now re-send `NewView` messages at each `consensus_timeout`. This neatly allows us to avoid implementing functionality for sending a message after a delay since restarted nodes will send their NewView message via this re-send mechanism :)